#!/usr/bin/env python
PACKAGE = "gradient_cost_plugin"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "Whether the plugin is enabled or not.", True);
gen.add("footprint_clearing_enabled", bool_t, 0, "If true, the robot footprint will clear (mark as free) the space in which it travels.", True);
gen.add("max_step", double_t, 0, "Maximum step allowed (in m), above which a cell is marked as LETHAL_OBSTACLE.", 0.2, 0.0, 2.0);
gen.add("max_angle", double_t, 0, "Maximum angle from horizontal (in degrees) of the landscape, above which the cell is marked as LETHAL_OBSTACLE.", 40.0, 0.0, 90.0);
gen.add("min_angle", double_t, 0, "Minimum angle from horizontal (in degrees) of the landscape, below which the cell is marked as FREE_SPACE.", 10.0, 0.0, 90.0);

exit(gen.generate(PACKAGE, "gradient_cost_plugin", "GradientCostPlugin"))